src/common/offline_transformations/src/pruning/shrink_weights.cpp

```diff
@@ -60,13 +59,25 @@ static bool maybe_adopt_reshape_node(std::shared_ptr<ov::Node> reshape, ngraph::
         return false;
     }
 
+    const auto new_shape = get_constant_from_source(shape)->cast_vector<int64_t>();
     auto sub_const_vector = std::vector<int64_t>();
```

Comment:
nitpick: you can pre-allocate the vector here for `mask->size()` size

Reply:
done